geode-for-redis/src/main/java/org/apache/geode/redis/internal/statistics/RedisStats.java

```diff
@@ -194,43 +201,49 @@ public void changeUniquePatternSubscriptions(long delta) {
 
   public void close() {
     geodeRedisStats.close();
-    stopPerSecondUpdater();
+    stopRollingAverageUpdater();
   }
 
-  private ScheduledExecutorService startPerSecondUpdater() {
-    int INTERVAL = 1;
+  private ScheduledExecutorService startRollingAverageUpdater() {
+    long microsPerSecond = 1_000_000;
+    final long delayMicros = microsPerSecond / rollingAverageSamplesPerSecond;
 
-    ScheduledExecutorService perSecondExecutor =
-        newSingleThreadScheduledExecutor("GemFireRedis-PerSecondUpdater-");
+    ScheduledExecutorService rollingAverageExecutor =
+        newSingleThreadScheduledExecutor("GemFireRedis-RollingAverageStatUpdater-");
 
-    perSecondExecutor.scheduleWithFixedDelay(
-        this::doPerSecondUpdates,
-        INTERVAL,
-        INTERVAL,
-        SECONDS);
+    rollingAverageExecutor.scheduleWithFixedDelay(this::doRollingAverageUpdates, delayMicros,
+        delayMicros, MICROSECONDS);
 
-    return perSecondExecutor;
+    return rollingAverageExecutor;
   }
 
-  private void stopPerSecondUpdater() {
-    perSecondExecutor.shutdownNow();
+  private void stopRollingAverageUpdater() {
+    rollingAverageExecutor.shutdownNow();
   }
 
-  private void doPerSecondUpdates() {
-    updateNetworkKilobytesReadLastSecond();
-    updateOpsPerformedOverLastSecond();
+  private void doRollingAverageUpdates() {
+    updateNetworkKilobytesReadLastSecond(rollingAverageTick);
+    updateOpsPerformedOverLastSecond(rollingAverageTick);
+    rollingAverageTick++;
+    if (rollingAverageTick >= rollingAverageSamplesPerSecond) {
+      rollingAverageTick = 0;
+    }
   }
 
-  private void updateNetworkKilobytesReadLastSecond() {
+  private void updateNetworkKilobytesReadLastSecond(int tickNumber) {
     final long totalNetworkBytesRead = getTotalNetworkBytesRead();
-    long deltaNetworkBytesRead = totalNetworkBytesRead - previousNetworkBytesRead;
-    networkKiloBytesReadOverLastSecond = deltaNetworkBytesRead / 1024.0;
-    previousNetworkBytesRead = totalNetworkBytesRead;
+    long deltaNetworkBytesRead = totalNetworkBytesRead - totalNetworkBytesReadLastTick;
```

Comment:
consider extracting this logic into a small class, RollingAverageStat that would have two fields: valueReadLastTick and valuesReadOverLastNSamples. It would have one method on it "long calculateRollingAverage(long)". You could call it like so: networkKiloBytesReadOverLastSecond = networkBytesRollingAverage.calculate(getTotalNetworkBytesRead()) / 1024.0 and also use it for opsPerformedOverLastSecond

Reply:
Nice suggestion, done